mmhuman3d/models/losses/prior_loss.py

```diff
@@ -65,6 +67,99 @@ def forward(self,
+        loss_weight (float, optional): The weight of the loss. Defaults to 1.0
+    """
+
+    def __init__(self,
+                 limb_convention,
+                 reduction='mean',
+                 loss_weight=1.0,
+                 eps=1e-4):
+        super().__init__()
+        assert reduction in (None, 'none', 'mean', 'sum')
+        self.reduction = reduction
+        self.loss_weight = loss_weight
+        self.eps = eps
+        self.limb_conf = None
+        self.limb_length = None
+        limb_idxs, _ = search_limbs(data_source=limb_convention)
+        limb_idxs = sorted(limb_idxs['body'])
+        self.limb_idxs = np.array(
+            list(x for x, _ in itertools.groupby(limb_idxs)))
+
+    def _get_limb_vec(self, keypoints3d):
+        kp_src = keypoints3d[:, self.limb_idxs[:, 0], :3]
+        kp_dst = keypoints3d[:, self.limb_idxs[:, 1], :3]
+        limb_vec = (kp_dst - kp_src).detach()
+        return limb_vec
+
+    def _compute_normed_limb(self, keypoints3d):
+        limb_vec = self._get_limb_vec(keypoints3d)
+        limb_length = torch.norm(limb_vec, dim=2, keepdim=True)
+        limb_vec_normed = limb_vec / (limb_length + self.eps)
+        return limb_vec, limb_vec_normed, limb_length
+
+    def forward(self,
+                keypoints3d_pred,
+                keypoints3d_target,
+                keypoints3d_pred_conf=None,
+                keypoints3d_target_conf=None,
+                loss_weight_override=None,
+                reduction_override=None):
+        """Forward function of LimbLengthLoss.
+
+        Args:
+            betas (torch.Tensor): The body shape parameters
+            loss_weight_override (float, optional): The weight of loss used to
+                override the original weight of loss
+            reduction_override (str, optional): The reduction method used to
+                override the original reduction method of the loss.
+                Defaults to None
+        Returns:
+            torch.Tensor: The calculated loss
+        """
+        assert reduction_override in (None, 'none', 'mean', 'sum')
+        reduction = (
+            reduction_override if reduction_override else self.reduction)
+        loss_weight = (
+            loss_weight_override
+            if loss_weight_override is not None else self.loss_weight)
+
+        if self.limb_conf is None:
+            # To compute self.limb_conf only once,
+            # we assume that target will not change during optimization
+            _, _, self.limb_length = self._compute_normed_limb(
+                keypoints3d_target)
+            self.limb_conf = torch.min(
+                keypoints3d_target_conf[:, self.limb_idxs[:, 1]],
+                keypoints3d_target_conf[:, self.limb_idxs[:, 0]])
+            self.limb_conf = self.limb_conf.unsqueeze(dim=-1)
+
+        pred_limb_vec, pred_limb_vec_normed, _ = self._compute_normed_limb(
+            keypoints3d_pred * keypoints3d_pred_conf.unsqueeze(dim=-1))
+
```

Comment:
`keypoints3d_pred` are regressed from SMPL and should have the same length anyway?

Reply:
In this case, why do we need this loss? Is this loss used as a soft shape constraint?

Reply:
This loss provides a way to optimize beta independently regardless of pose.